packages/react-jsx-highcharts/src/components/Series/Series.js

```diff
@@ -95,7 +97,23 @@ const Series = memo(
 
       const modifiedProps = getModifiedProps(prevProps, rest);
       if (modifiedProps !== false) {
-        series.update(modifiedProps, false);
+        const nonEventProps = getNonEventHandlerProps(modifiedProps);
+        series.update(nonEventProps, false);
+
+        // update changed eventhandlers
+        const modifiedEvents = getEventsConfig(modifiedProps);
+        const prevEvents = getEventsConfig(prevProps);
+        Object.keys(modifiedEvents).forEach(eventName => {
```

Comment:
This works really well, apart from the in the case of #252. If an event handler is removed completely, the handler itself won't be removed.

Current this test case fails
```js
it('removes onHide eventhandler when prop is omitted', () => {
  const onHide = jest.fn();

  const wrapper = render(<Component onHide={onHide} />);

  wrapper.rerender(<Component />);

  seriesRef.current.object.hide();
  expect(onHide).not.toHaveBeenCalled();
});
```

Obviously that is existing broken functionality, but I'm wondering if it's something we need to revisit?

Reply:
When I tried to solve #252, I found out that some properties cannot be set to null with chart.update. They should be set to their default values when omitted from props.

I cannot remember the exact details, but for example:
- [enableMouseTracking](https://api.highcharts.com/highcharts/series.line.enableMouseTracking) default value is true, setting it to null would make it falsy.
- [tooltip](https://api.highcharts.com/highcharts/series.line.tooltip) is an object property, setting that to null can have strange side effects

So when omitting props this library would need to know what to set in every property.

I just gave up when I realised that.

I think that for now it is better to have consistently broken functionality rather than fixing it for event handlers only.

Thoughts?

Reply:
My only thoughts are if we can somehow utilise [`Highcharts.defaultOptions`](https://api.highcharts.com/class-reference/Highcharts#.defaultOptions), but that would likely be a breaking change due to unforeseen consequences.

I've been getting more comfortable with Typescript recently, whilst contributing to [Remirror](https://www.github.com/remirror/remirror), so thinking more and more about a Typescript rewrite, maybe something to consider then?

Reply:
Typescript can be setup for gradual upgrade too. If you want, I can do that when I have the time.

Should we merge this as is?